tests/test_2024_04_23.py

```diff
@@ -0,0 +1,132 @@
+        vote_id, _ = start_vote(tx_params, silent=True)
+
+    vote_tx = helpers.execute_vote(accounts, vote_id, contracts.voting)
+
+    print(f"voteId = {vote_id}, gasUsed = {vote_tx.gas_used}")
+
+    #old GateSeal permissions after
+    _check_no_role(contracts.withdrawal_queue, "PAUSE_ROLE", old_gate_seal)
+    _check_no_role(contracts.validators_exit_bus_oracle, "PAUSE_ROLE", old_gate_seal)
+    #new GateSeal permissions after
+    _check_role(contracts.withdrawal_queue, "PAUSE_ROLE", new_gate_seal)
+    _check_role(contracts.validators_exit_bus_oracle, "PAUSE_ROLE", new_gate_seal)
+
+    #Асceptance test
+    new_gate_seal_contract = interface.GateSeal(new_gate_seal)
+    assert new_gate_seal_contract.get_sealing_committee() == GATE_SEAL_COMMITTEE
+    sealables = new_gate_seal_contract.get_sealables()
+    assert len(sealables) == 2
+    assert contracts.validators_exit_bus_oracle.address in sealables
+    assert contracts.withdrawal_queue.address in sealables
+
+    assert new_gate_seal_contract.get_seal_duration_seconds() == GATE_SEAL_PAUSE_DURATION
+    assert new_gate_seal_contract.get_expiry_timestamp() == NEW_EXPIRY_TIMESTAMP
+    assert not new_gate_seal_contract.is_expired()
+
+    #Scenario test
+    print(f"Simulating GateSeal flow")
+
+    #Try to use the Old gate seal to pause the contracts
+    print("Try to use the Old gate seal to pause the contracts")
+    print(chain.time())
+    with reverts("10"): # converted into string list of sealed indexes (in sealables) in which the error occurred, in the descending order
+        contracts.gate_seal.seal(sealables, {"from": GATE_SEAL_COMMITTEE})
+
+    print("Seal the contracts with the New gate seal")
+    new_gate_seal_contract.seal(sealables, {"from": GATE_SEAL_COMMITTEE})
+    print("Sealed")
+
+    expiry_timestamp = chain.time()
+    assert new_gate_seal_contract.is_expired()
+    assert expiry_timestamp - new_gate_seal_contract.get_expiry_timestamp() <= 2
+    print("Expired")
+
+    for sealable in sealables:
+        assert interface.IPausable(sealable).isPaused()
+    print("Sealables paused")
+    chain.sleep(6 * 60 * 60 * 24)
+    chain.mine()
+
+    for sealable in sealables:
+        assert not interface.IPausable(sealable).isPaused()
+    print(f"Sealables unpaused in {new_gate_seal_contract.get_seal_duration_seconds()}")
+
+    print("GateSeal is good to go!")
+
+    # Validating events
+    assert count_vote_items_by_events(vote_tx, contracts.voting) == 4, "Incorrect voting items count"
+
+    metadata = find_metadata_by_vote_id(vote_id)
+    assert get_lido_vote_cid_from_str(metadata) == "bafkreihn3cl5nzzgnng3bhcwpu7lk6iykniykdjfcr5hexdydjt3lqt5k4"
+
+    display_voting_events(vote_tx)
+
+    if bypass_events_decoding or network_name() in ("goerli", "goerli-fork"):
+        return
+
+    evs = group_voting_events(vote_tx)
+
+    # Grant PAUSE_ROLE on WithdrawalQueue for the new GateSeal
+    validate_grant_role_event(evs[0], PAUSE_ROLE, new_gate_seal, agent)
```

Comment:
it would be nice to have check, that emitter is target contract

Reply:
technical debt